Probability  is the branch of mathematics concerning numerical descriptions of how likely an event is to occur, or how likely it is that a proposition is true. The probability of an event is a number between 0 and 1, where, roughly speaking, 0 indicates impossibility of the event and 1 indicates certainty. The higher the probability of an event, the more likely it is that the event will occur (https://en.wikipedia.org/wiki/Probability)

$$ P(A) = \frac{n(A)}{n(S)}$$

                                0 <= P(A) <= 1 or 0% <= P(A) <= 100%

Basic definition:

+ Set - a collection of elements or objects of interest

+ Empty set (denoted by ∅) a set containing no elements 

+ Universal set (denoted by S) a set containing all possible elements 

+ Complement (Not). The complement of A is  a set containing all elements of S not in A
$$(A' or A^c)$$

+ Intersection (And) (A ∩ B) – a set containing all elements in both A and B 

+ Union (Or) (A u B) – a set containing all elements in A or B or both 

+ Mutually Exclusive or Disjoint Sets: A n B = 0




Basic formula:

+ $$P(A n B) = \frac{n(A n B)}{n(S)}$$

+ $$P(A^c) = 1 - P(A)$$

+ $$P(A u B) = \frac{n(A u B)}{n(S)} = P(A) + P(B) - P(A n B)$$

+ Conditional probability of A given B:
$$P(A|B) = \frac{P(A n B)}{P(B)}$$
                        where P(B) not equal to 0 (P(B) != 0)

Exercise 1: What is the probability of rolling a dice and its value is less than 4 knowing that the value is an odd number?

In [8]:
# Let A is the event that a dice after rolling is less than 4 and an odd number
# A = {1,3}
A = {1,3}
nA = len(A) # Count how many numbers in set A

# Let S is the universal set that the dice value is an odd number
S = {1,3,6}
nS =len(S) # Count how many numbers in set B

# Compute the probility of event A
probA = nA/nS

# Print output
print(probA)

0.6666666666666666


The answer is 0.667 or 66.7% 

Exercise 2: At the International University, the probability that a student takes Computer Programming and Spanish is 0.15. The probability that a student takes Computer Programming is 0.4.

What is the probability that a student takes Spanish given that the student is taking Computer Programming?


In [12]:
# First, summarize and denote
# Let A be the event that a student takes Computer Programming
CP = 0.4

# Let B be the event that a student takes Computer Programming and Spanish is 0.15
CPnSp = 0.15

# Compute the probability that a student takes Spanish given that the student is taking Computer Programming
# P(B|A) = P(AnB) / P(A)
prob = CPnSp / CP

# Print output
print(prob)




0.37499999999999994


The probability that a student takes Spanish given that the student is taking 
Computer Programming is 37.5%

Exercise 3: Here are the results of a survey completed with adult parents with children. 

 

|   | Too Expensive  | Affordable  | Too cheap  |   
|---|---|---|---|
|  Child in college | 55%  | 4%  | 0%  |   
|  Child not in college | 30%  | 8%  | 3%  |   


What is the probability a person thinks college is too expensive given they have a child in
college?

P(Child in College) = P(CiC) = 55% + 4% + 0% = 59%

P(CiC n TooExp) = 55%

$$P(TooExp | CiC) = \frac{0.55}{0.59} = 0.932$$


Exercise 4: Forty seven mice were divided according to species and nutrition levels as follows:



|   | High nutrition (H)  | Low Nutrition (L)  |   
|---|---|---|
|  A | 15  | 10  |   
|  B | 9  | 13  | 



a/ Find P(B u L)

b/ Find P(A|H)

c/ Are events A and H independent on each other?

The value in the provided table were not in forms of probability

Thus, first, we have to take each number and divide by 47 to find probability

Then, we can use those value after division to answer the questions



|   | High nutrition (H)  | Low Nutrition (L)  |   
|---|---|---|
|  A | 15/47  | 10/47  |   
|  B | 9/47  | 13/47  | 



a/ P(B) = 9/47 + 13/47 = 22/47

P(L) = 10/47 + 13/47 = 23/47

P(B n L) = 13/47

P (B u L ) = P(B) + P(L) - P(B n L) = 22/47 + 23/47 - 13/47 = 32/47 = 0.681

In [14]:
pB = 9/47 + 13/47
pL = 10/47 + 13/47
pBnL = 13/47

pBuL = pB + pL - pBnL

print(pBuL)

0.6808510638297873


b/ p(H) = 15/47 + 9/47 = 24/47

p(A n H) = 15/47

p(A|H) = P(A n H) / P(H) = (15/47) / (24/47) = 15/24 = 0.625


In [18]:
pH = 15/47 + 9/47

pAnH = 15/47

pA_given_H = pAnH / pH

print(pA_given_H)



0.625


c/ Events A and H are independent if the equation P(A n H) = P(A) * P(H) holds true

p(A n H) = 15/47

p(A)*p(H) = 25/47 * 24/47 = 600/2209 

Since p(A n H) does not equal p(A)*p(H)

Thus, A and H are not independent events